### PR TITLE
Add pinot docs for HyperLogLogPlus

### PR DESCRIPTION
**Pull request by deemoliu**
Add pinot docs for hyperLogLogPlus functions.

https://github.com/apache/pinot/pull/11346
https://github.com/apache/pinot/pull/11747

### CODE REVIEW

**Comment by deemoliu** [comment]
cc: @Jackie-Jiang

**Comment by Jackie-Jiang** [comment]
cc @kelseiv

**Comment by kelseiv** [comment]
thank you, @deemoliu @Jackie-Jiang! 
for awareness, when we're adding new docs via GitHub, we'll need to add a reference to the new doc in the [SUMMARY.md ](https://github.com/pinot-contrib/pinot-docs/blob/latest/SUMMARY.md)file to be able to properly link to the doc. 

in this case specifically, for functions, in the [Configuration Reference section](https://github.com/pinot-contrib/pinot-docs/blob/latest/SUMMARY.md#configuration-reference)

File: configuration-reference/functions/distinctcounthllplusmv.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
+---
+description: >-
+  This section contains reference documentation for the DISTINCTCOUNTHLLPLUSMV
+  function.
+---
+
+# DISTINCTCOUNTHLLPLUSMV
+
+Returns an approximate distinct count using HyperLogLogPlusPlus in a group
```
Comment (kelseiv) [inline configuration-reference/functions/distinctcounthllplusmv.md]:
```suggestion
Returns an approximate distinct count using HyperLogLogPlusPlus in a group.
```

File: configuration-reference/functions/distinctcountrawhllplusmv.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
+---
+description: >-
+  This section contains reference documentation for the DISTINCTCOUNTRAWHLLPLUSMV
+  function.
+---
+
+# DISTINCTCOUNTRAWHLLPLUSMV
+
+Returns HLLPlus response serialized as string. The serialized HLLPlus can be converted back into an HLLPlus and then aggregated with other HLLPluses. A common use case may be to merge HLLPlus responses from different Pinot tables, or to allow aggregation after client-side batching.
```
Comment (kelseiv) [inline configuration-reference/functions/distinctcountrawhllplusmv.md]: to be sure, is there any distinction to make between this description and the description for DISTINCTCOUNTRAWHLLPLUS?

File: users/user-guide-query/how-to-handle-unique-counting.md [hunk 3]
```diff
@@ -26,11 +26,33 @@ Functions:
 
 * **DistinctCountHLL**(x)\_ -> LONG\_
 
-For column type **INT**/**LONG**/**FLOAT**/**DOUBLE**/**STRING** , Pinot treats each value as an individual entry to add into HyperLogLog Object, then compute the approximation by calling method **cardinality()**.
+For column type **INT**/**LONG**/**FLOAT**/**DOUBLE**/**STRING** , Pinot treats each value as an individual entry to add into HyperLogLog Object, then compute the approximation by calling method _**cardinality()**.
```
Comment (kelseiv) [inline users/user-guide-query/how-to-handle-unique-counting.md]:
```suggestion
For column type **INT**/**LONG**/**FLOAT**/**DOUBLE**/**STRING**, Pinot treats each value as an individual entry to add into HyperLogLog Object, and then computes the approximation by calling method _**cardinality()**.
```

File: users/user-guide-query/how-to-handle-unique-counting.md [hunk 4]
```diff
@@ -26,11 +26,33 @@ Functions:
...
 
 For column type **BYTES**, Pinot treats each value as a serialized HyperLogLog Object with pre-aggregated values inside. The bytes value is generated by `org.apache.pinot.core.common.ObjectSerDeUtils.HYPER_LOG_LOG_SER_DE.serialize(hyperLogLog)`.
 
-All deserialized HyperLogLog object will be merged into one then calling method _\*\*cardinality() \*\*to get the approximated unique count._
+All deserialized HyperLogLog object will be merged into one then calling method _**cardinality()** to get the approximated unique count._
```
Comment (kelseiv) [inline users/user-guide-query/how-to-handle-unique-counting.md]: to verify, can there be multiple object here, or is Object a proper term in the context of HyperLogLog? (see it's capitalized in line 29)

Comment (kelseiv) [reply users/user-guide-query/how-to-handle-unique-counting.md]: would rephrasing as follows be accurate?:

All deserialized HyperLogLog objects are merged into one, and then Pinot calls the _**cardinality()** method to get the approximated unique count.

File: users/user-guide-query/how-to-handle-unique-counting.md [hunk 5]
```diff
@@ -26,11 +26,33 @@ Functions:
...
+
+### HyperLogLogPlusPlus
+
+The [HyperLogLog++](https://en.wikipedia.org/wiki/HyperLogLog) algorithm proposes several improvements in the HyperLogLog algorithm to reduce memory requirements and increase accuracy in some ranges of cardinalities.
+
+* 64-bit hash function is used instead of the 32 bits used in the original paper. This reduces the hash collisions for large cardinalities allowing to remove the large range correction.
```
Comment (kelseiv) [inline users/user-guide-query/how-to-handle-unique-counting.md:39]: in the original algorithm?

Comment (kelseiv) [reply users/user-guide-query/how-to-handle-unique-counting.md:39]: would this be accurate:
This reduces the hash collisions for large cardinalities, removing the large range correction.

File: users/user-guide-query/how-to-handle-unique-counting.md [hunk 6]
```diff
@@ -26,11 +26,33 @@ Functions:
...
+* Some bias is found for small cardinalities when switching from linear counting to the HLL counting. An empirical bias correction is proposed to mitigate the problem.
+* A sparse representation of the registers is proposed to reduce memory requirements for small cardinalities, which can be later transformed to a dense representation if the cardinality grows.
```
Comment (kelseiv) [inline users/user-guide-query/how-to-handle-unique-counting.md]: is proposed >> is implemented?

File: users/user-guide-query/how-to-handle-unique-counting.md [hunk 7]
```diff
@@ -26,11 +26,33 @@ Functions:
...
+
+Pinot leverages [HyperLogLogPlus Class](https://github.com/addthis/stream-lib/blob/master/src/main/java/com/clearspring/analytics/stream/cardinality/HyperLogLogPlus.java) in library `com.clearspring.analytics:stream:2.7.0`as the data structure to hold intermediate results.
+
+Functions:
+
+* **DistinctCountHLLPlus**(*\<HllPlusColumn>*)\_ -> LONG\_
+* **DistinctCountHLLPlus**(*\<HllPlusColumn>, \<p>*)\_ -> LONG\_
+* **DistinctCountHLLPlus**(*\<HllPlusColumn>, \<p>, \<sp>*)\_ -> LONG\_
+
+For column type **INT**/**LONG**/**FLOAT**/**DOUBLE**/**STRING** , Pinot treats each value as an individual entry to add into HyperLogLogPlus Object, then compute the approximation by calling method _**cardinality()**.
```
Comment (kelseiv) [inline users/user-guide-query/how-to-handle-unique-counting.md]: ditto

File: users/user-guide-query/how-to-handle-unique-counting.md [hunk 8]
```diff
@@ -26,11 +26,33 @@ Functions:
...
+
+For column type **BYTES**, Pinot treats each value as a serialized HyperLogLogPlus Object with pre-aggregated values inside. The bytes value is generated by `org.apache.pinot.core.common.ObjectSerDeUtils.HYPER_LOG_LOG_PLUS_SER_DE.serialize(hyperLogLogPlus)`.
+
+All deserialized HyperLogLogPlus object will be merged into one then calling method _**cardinality()** to get the approximated unique count._
```
Comment (kelseiv) [inline users/user-guide-query/how-to-handle-unique-counting.md]: ditto

**Review by kelseiv — COMMENTED** [review]
minor edits

File: configuration-reference/functions/distinctcountrawhllplus.md [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+---
+
+# DISTINCTCOUNTRAWHLLPLUS
+
+Returns HLLPlus response serialized as string. The serialized HLLPlus can be converted back into an HLLPlus and then aggregated with other HLLPluses. A common use case may be to merge HLLPlus responses from different Pinot tables, or to allow aggregation after client-side batching.
+
+## Signature
+
+> DISTINCTCOUNTRAWHLLPLUS(colName)
+> DISTINCTCOUNTRAWHLLPLUS(colName, p)
```
Comment (cbalci) [inline configuration-reference/functions/distinctcountrawhllplus.md:15]: Should we name these parameters more explicitly? I believe these refer to 'precision' and 'sparse? precision', would be good to have some basic explanation here.

File: users/user-guide-query/how-to-handle-unique-counting.md (hunk 4 above)
Comment (deemoliu) [reply users/user-guide-query/how-to-handle-unique-counting.md]: i am ok with both this line or the original statement. actually i only made a small change to fix the format of this line.

File: configuration-reference/functions/distinctcountrawhllplusmv.md (hunk 2 above)
Comment (deemoliu) [reply configuration-reference/functions/distinctcountrawhllplusmv.md]: yes. there is not much difference between the DISTINCTCOUNTRAWHLLPLUS and DISTINCTCOUNTRAWHLLPLUSMV.

File: users/user-guide-query/how-to-handle-unique-counting.md (hunk 5 above)
Comment (deemoliu) [reply users/user-guide-query/how-to-handle-unique-counting.md:39]: the wiki link for HyperLogLog https://en.wikipedia.org/wiki/HyperLogLog, it has a section pointing to HyperLogLog++ algorithm. the rephrase looks ok.

**Comment by deemoliu** [comment]
@kelseiv updated, please take a look.

File: configuration-reference/functions/distinctcountrawhllplus.md (hunk 9 above)
Comment (deemoliu) [reply configuration-reference/functions/distinctcountrawhllplus.md:15]: @cbalci i added a one line in the above section to describe p and sp, please take a look, thanks

**Review by kelseiv — APPROVED** [review]

**Pull request merged**